src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt

```diff
@@ -41,15 +44,16 @@ class AnonymousElectorDocumentService(
     @Transactional
     fun reIssueAnonymousElectorDocument(eroId: String, dto: ReIssueAnonymousElectorDocumentDto): PdfFile {
         val gssCodes = eroService.lookupGssCodesForEro(eroId)
-        val mostRecentAed = anonymousElectorDocumentRepository.findByGssCodeInAndSourceTypeAndSourceReference(
-            gssCodes = gssCodes,
-            sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
-            sourceReference = dto.sourceReference
-        )
-            .sortedByDescending { it.dateCreated }
+        val mostRecentAed = getAnonymousElectorDocumentsSortedByDate(gssCodes, dto.sourceReference)
             .firstOrNull() ?: throw CertificateNotFoundException(eroId, ANONYMOUS_ELECTOR_DOCUMENT, dto.sourceReference)
 
-        TODO("not yet implemented")
+        val templateFilename = pdfTemplateDetailsFactory.getTemplateFilename(mostRecentAed.gssCode)
+        with(reIssueAnonymousElectorDocumentMapper.toNewAnonymousElectorDocument(mostRecentAed, templateFilename)) {
+            val templateDetails = pdfTemplateDetailsFactory.getTemplateDetails(this)
+            val contents = pdfFactory.createPdfContents(templateDetails)
+            anonymousElectorDocumentRepository.save(this)
```

Comment:
Is there any possibility to refactor this logic with existing generateAed service function?

Reply:
Almost certainly, yes 👍 
I'll take a look 👍

Reply:
Done 👍